examples_gbs/run_tutorial_dense.py

```diff
@@ -0,0 +1,120 @@
+# pylint: disable=wrong-import-position,wrong-import-order,ungrouped-imports
+"""
+Dense Subgraph Tutorial
+=========================
+Graphs can be used to model a wide variety of objects: social networks, financial markets,
+biological networks, and many others. A common problem of interest is to find subgraphs that
+contain a large number of connections between its nodes. These may correspond to communities in
+social networks, correlated assets in a market, or mutually influential proteins in a biological
+network. This tutorial studies how GBS can be used to identify dense subgraphs.
+
+Mathematically, this task is known as the `dense subgraph problem
+<https://en.wikipedia.org/wiki/Dense_subgraph>`_. The density of a graph is equal to the number of

```

Comment:
@josh146, I still never got my head around whether we should have a single or double underscore here?

Reply:
Me too! What's the mnemonic so we never forget?

Reply:
Here are some answers:
https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use
http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#anonymous-hyperlinks

So a double underscore is for a "one off" link, whereas a single allows for that link to be used again, e.g. from the second reference above:

```rst
See the `Python home page <http://www.python.org>`_ for info.
This `link <Python home page_>`_ is an alias to the link above.
```

So I guess it's kind of like whether we add a `\label` in latex.

In our case we would do
```rst
Mathematically, this task is known as the `dense subgraph problem <https://en.wikipedia.org/wiki/Dense_subgraph>`_.
Another link to the `dense subgraph problem <dense subgraph problem_>`
```

Which is actually ok, although I'm not sure if we'd ever remember when working in another document that we've already provided the link. So for now have removed.